### DISCUSSION TITLE
Difficulties importing certificate

### DISCUSSION THREAD

**Discussion by camielwnzl** [PaloAltoNetworks/terraform-provider-panos#360] [Q&A]
I'm attempting to import a certificate and key into Panorama using the:

resource "panos_certificate_import" "cert" {
  name     = "sample_cert
  template = var.panos_device_template
  pkcs12 {
    certificate = file("${path.module}/certs/cert.pfx")
    passphrase  = "changeit"
  }
  lifecycle {
    create_before_destroy = true
  }
}

However - i'm getting: Call to function "file" failed: contents of "./certs/cert.pfx" are not valid UTF-8; use the filebase64 function to obtain the Base64 encoded
│ contents or the other file functions (e.g. filemd5, filesha256) to obtain file hashing results instead.

Using filebase64 I get: Error: Import of sample_cert_pem failed. Failed to extract certificate

I'm able to import the certificate and key using the web UI.

Are you able to assist ?

**Comment by shinmog** [comment]
I think behavior of the `file()` function changed between when this resource was created and right now...

Looking at the Terraform docs, they advocate for providers to use `filebase64()` to read in binary data, and have the provider take in the base64 data and decode that.  So this resource will need a new parameter to handle this, allowing users to specify one or the other, and doing the base64 decode within the resource.

For now if it's possible, use PEM format instead since that data is UTF-8 and can be read by `file()` just fine.
